### PR TITLE
chore: Dockerfile/ci fixes

### PR DESCRIPTION
**Pull request by pjenvey**
- fix the final RUN command
- include ssl for rusoto/reqwest and upgrade to latest rust
- possibly fix branch triggers (or is circleci just acting up)?
- remove unnecessary WITH_RUST to travis

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,8 +8,7 @@ RUN \
     cargo --version && \
     rustc --version && \
     mkdir -m 755 bin && \
-    cargo build --release && \
-    cp /app/target/release/autopush_rs /app/bin
+    cargo install --root /app
```
Comment (bbangert) [inline Dockerfile]: I copied this from pushbox, so that prolly needs a similar fix.

Comment (pjenvey) [reply Dockerfile]: no fix here, just a 'refactor'. pushbox ran into a nice rust-nightly bug requiring this workaround for a few days

Comment (jrconlin) [reply Dockerfile]: I should check if that bug is still in play, TBH.

Comment (pjenvey) [reply Dockerfile]: I've verified it's fixed as of megaphone's pegged nightly-2018-05-05 or somewhere near there

**Pull request merged**